src/disk_manager/scalog_replication_manager.cc

```diff
@@ -199,6 +192,24 @@ class ScalogReplicationServiceImpl final : public ScalogReplicationService::Serv
 			}
 		}
 
+		void WriterLoop() {
+			while (running_) {
+				WriteTask task(ScalogReplicationRequest{});
+				if (!write_queue_.read(task)) continue;
+
+				std::lock_guard<std::mutex> file_lock(file_mutex_);
```

Comment:
If multiple WriterLoop is run by multiple threads, this lock will serialize them. So I think we need to unlock it before WriteRequest